# HTGAA Bio-Bootcamp Day 3

## Coding Basics, DNA Designs, Genetic Circuits, Accessible Biolabs

## Date: Friday January 23rd, 2026 9 AM \- 12 AM EST

This handout is an overview about the topics we will cover on the first day of Bio-Bootcamp and resources that you can explore ahead of or following the Bio-Bootcamp.

## Summary: 

The third session dives into coding and software tools for bioinformatics. Participants will practice Python fundamentals, explore Google Colab for biological data analysis, and learn how to perform *in silico* cloning using Benchling.

## Day 3 Topics & Agenda

1. ### Introduction

   * ***Speaker*****:** Ethan Itovitch  
   * ***Subtopics:***   
     * Welcome   
     * Today‚Äôs learning objectives

2. ### Python Fundamentals & Colab

   * *Speaker:* Dr. Paul Kao  
   * *Subtopics:*   
     * Fundamentals of Python  
     * Basic Syntax  
     * Setting up Colab

3. ### Benchling Basics

   * *Speaker:* Dr. Cholpisit (Ice) Kiattisewee  
   * *Subtopics:*   
     * Reading DNA sequences  
     * *in silico* cloning by scenario  
     * essential Benchling features

4. ### Accessible Biolabs

   * *Speaker:* Adrian Filips  
   * *Subtopics:*   
     * Setting up labs with limited resources  
     * accessible protocols for external students

5. ### Colab Examples and AI Tools

   * *Speaker:* Suvin Sundararajan  
   * *Subtopics:*   
     * From Code to Lab: Practical Applications with CoLab & Opentrons

6. ### HTGAA Infrastructure

   * *Speakers:* Greg Galparin  
   * *Subtopics:*   
     * HTGAA Website \- creating your page

## Day 3 Resources

### Supporting Colabs:

* **Walkthrough:** [Python-Basics-HTGAA-Bootcamp-2025.ipynb](https://colab.research.google.com/drive/1-94Ws4yLD5KlFYi3JSmLnpPADaWxKPB-?usp=sharing)  
* **Example 1: Rosetta:** [(Remake) \- L-Protein \- Point-Mutation-Scan.ipynb](https://colab.research.google.com/drive/1Z5rw9CJnZ1EXxeKv_xPzv1_nV7BXhnRZ?usp=sharing)  
* **Example 2: Opentrons Lab:**	[HTGAA25 Opentrons Colab](https://colab.research.google.com/drive/1VoouRH0nqlk09g50rHxOElaLD-SVknYY#scrollTo=MNdqL8tjfa83)  
* **Part 2:** [Python-HTGAA-Bootcamp-2025.ipynb](https://colab.research.google.com/drive/1frkZX-XXTdhHTIy9qAo4LjrHbI7qZD4P#scrollTo=oc3_TY7hGOm3)  
* **Part 2 Opentrons:**[HTGAA25 Opentrons Bare Code](https://colab.research.google.com/drive/14kRcUqYPEGdaVrXuRLJbMQw4QLia65fg#scrollTo=KcLYt3JFg1iU)

### AI tools:

* **üìù**Connected Papers: [https://www.connectedpapers.com/](https://www.connectedpapers.com/)  
* **üìù**Research Rabbit: [https://www.researchrabbit.ai/](https://www.researchrabbit.ai/)

### Benchling Example Project:

* **üìù** [Benchling Basics](https://benchling.com/cholpisit-k/f_/QZCBnlIr-benchling-basics/)

### Colab Examples and AI Tools Talk Resources

* **Safety, biosafety, and regulatory anchors:**  
  * NIH Guidelines PDF (Apr 2024\) [https://osp.od.nih.gov/wp-content/uploads/](https://osp.od.nih.gov/wp-content/uploads/NIH_Guidelines.pdf)  
    [NIH\_Guidelines.pdf](https://osp.od.nih.gov/wp-content/uploads/NIH_Guidelines.pdf)  
  * NIH OSP Biosafety & Biosecurity Policy: [https://osp.od.nih.gov/policies](https://osp.od.nih.gov/policies/biosafety-and-biosecurity-policy/)  
    [/biosafety-and-biosecurity-policy/](https://osp.od.nih.gov/policies/biosafety-and-biosecurity-policy/)  
  * CDC BMBL hub: [https://www.cdc.gov/labs/bmbl/index.html](https://www.cdc.gov/labs/bmbl/index.html)  
* **DNA design \+ assembly planning:**  
  * NEBuilder Assembly Tool: [https://nebuilder.neb.com/](https://nebuilder.neb.com/)  
  * NCBI (PubMed/NCBI portal): [https://static.pubmed.gov/portal/portal.fcgi/](https://static.pubmed.gov/portal/portal.fcgi/)  
  * NCBI BLAST: [https://blast.ncbi.nlm.nih.gov/Blast.cgi](https://blast.ncbi.nlm.nih.gov/Blast.cgi)  
  * IDT OligoAnalyzer: [https://www.idtdna.com/pages/tools/oligoanalyzer](https://www.idtdna.com/pages/tools/oligoanalyzer)  
  * IDT Codon Opt: [https://www.idtdna.com/pages/tools/codon-optimization-tool](https://www.idtdna.com/pages/tools/codon-optimization-tool)  
  * Twist Codon Optimization: [https://www.twistbioscience.com](https://www.twistbioscience.com/faq/using-your-twist-account/what-does-twist-codon-optimization-tool-do)  
    [/faq/using-your-twist-account/what-does-twist-codon-optimization-tool-do](https://www.twistbioscience.com/faq/using-your-twist-account/what-does-twist-codon-optimization-tool-do)  
* **Advanced:**   
  * Plasmid screening: [https://blog.addgene.org/plasmids-101-plasmid-screening-strategies](https://blog.addgene.org/plasmids-101-plasmid-screening-strategies)  
  * Molecular Biology Reference: https://www.addgene.org/mol-bio-reference/  
* **Protocol development, reproducibility, and sharing**  
  * protocols.io: [https://www.protocols.io/](https://www.protocols.io/)  
  * Benchling: [https://www.benchling.com/](https://www.benchling.com/)  
  * Opentrons Protocol Library: [https://library.opentrons.com/](https://library.opentrons.com/)

* **Useful resources borrowed from DNA Read Write Edit**  
  * DNA Sequencing at 40: Past, Present, and Future (2017) Shendure, J., Balasubramanian, S., Church, G. *et al.* [https://doi.org/10.1038/nature24286](https://doi.org/10.1038/nature24286) DNA   
  * Synthesis Technologies to Close the Gene Writing Gap (2023), Hoose, A., Vellacott, R., Storch, M. *et al.* [https://doi.org/10.1038/s41570-022-00456-9](https://doi.org/10.1038/s41570-022-00456-9)   
  * Recombineering and MAGE (2021), Wannier T, et al. Nat Rev Methods Primers, [https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9083505/](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9083505/)  
  * CRISPR Technology: A Decade of Genome Editing is Only the Beginning, Wang, Doudna, *et al.,* [https://www.science.org/doi/10.1126/science.add8643](https://www.science.org/doi/10.1126/science.add8643)

